src/julia-syntax.scm

```diff
@@ -2118,11 +2118,23 @@
                      `(call ,@hvncat ,dims ,(tf is-row-first) ,@aflat))
                     `(call ,@hvncat ,(tuplize shape) ,(tf is-row-first) ,@aflat))))))))
 
-(define (expand-property-destruct lhss x)
-  (if (not (length= lhss 1))
-      (error (string "invalid assignment location \"" (deparse lhs) "\"")))
-  (let* ((xx (if (symbol-like? x) x (make-ssavalue)))
-         (ini (if (eq? x xx) '() (list (sink-assignment xx (expand-forms x))))))
+(define (maybe-ssavalue lhss x in-lhs?)
+  (cond ((or (and (not (in-lhs? x lhss)) (symbol? x))
+             (ssavalue? x))
+          x)
+        ((and (pair? lhss) (vararg? (last lhss))
+              (eventually-call? (cadr (last lhss))))
+         (gensy))
+        (else (make-ssavalue))))
+
+(define (expand-property-destruct lhs x)
+  (if (not (length= lhs 1))
+      (error (string "invalid assignment location \"" (deparse `(tuple ,lhs)) "\"")))
+  (define (in-lhs? x lhss)
+    (any (lambda (l) (eq? l x)) lhss))
```

Comment:
```suggestion
    (memq x lhss))
```

Reply:
Don't know why I didn't think of that. Really no need to define an extra closure in that case.